Gems/Atom/RPI/Code/Source/RPI.Public/Model/ModelTagSystemComponent.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+/**
+* @file ModelTagSystemComponent.cpp
+* @brief Contains the definition of the ModelTagSystemComponent methods that aren't defined as inline
+*/
+
+#include <Atom/RPI.Public/Model/ModelTagSystemComponent.h>
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/std/sort.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        ModelTagSystemComponent::~ModelTagSystemComponent() = default;
```

Comment:
It looks like the implementation of ImageTagSystemCompnent and ModelTagSystemComponent are almost same. It would be nice we only need one AssetTagSystem for the implementation and use it for any type of asset. 
It's not required in this PR.